src/subclass/io_stream.rs

```diff
@@ -0,0 +1,162 @@
+
+    fn parent_get_output_stream(&self, stream: &IOStream) -> crate::OutputStream;
+
+    fn parent_close(
+        &self,
+        stream: &IOStream,
+        cancellable: Option<&Cancellable>,
+    ) -> Result<(), Error>;
+}
+
+impl<T: IOStreamImpl + ObjectImpl> IOStreamImplExt for T {
+    fn parent_get_input_stream(&self, stream: &IOStream) -> crate::InputStream {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gio_sys::GIOStreamClass;
+            let f = (*parent_class)
+                .get_input_stream
+                .expect("No parent class implementation for \"get_input_stream\"");
+            from_glib_none(f(stream.to_glib_none().0))
+        }
+    }
+
+    fn parent_get_output_stream(&self, stream: &IOStream) -> crate::OutputStream {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gio_sys::GIOStreamClass;
+            let f = (*parent_class)
+                .get_output_stream
+                .expect("No parent class implementation for \"get_output_stream\"");
+            from_glib_none(f(stream.to_glib_none().0))
+        }
+    }
+
+    fn parent_close(
+        &self,
+        stream: &IOStream,
+        cancellable: Option<&Cancellable>,
+    ) -> Result<(), Error> {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gio_sys::GIOStreamClass;
+            let mut err = ptr::null_mut();
+            if let Some(f) = (*parent_class).close_fn {
+                if from_glib(f(
+                    stream.to_glib_none().0,
+                    cancellable.to_glib_none().0,
+                    &mut err,
+                )) {
+                    Ok(())
+                } else {
+                    Err(from_glib_full(err))
+                }
+            } else {
+                Ok(())
+            }
+        }
+    }
+}
+
+unsafe impl<T: ObjectSubclass + IOStreamImpl> IsSubclassable<T> for IOStreamClass {
+    fn override_vfuncs(&mut self) {
+        <glib::ObjectClass as IsSubclassable<T>>::override_vfuncs(self);
+        unsafe {
+            let klass = &mut *(self as *mut Self as *mut gio_sys::GIOStreamClass);
+            klass.get_input_stream = Some(stream_get_input_stream::<T>);
+            klass.get_output_stream = Some(stream_get_output_stream::<T>);
+            klass.close_fn = Some(stream_close::<T>);
+        }
+    }
+}
+
+unsafe extern "C" fn stream_get_input_stream<T: ObjectSubclass>(
+    ptr: *mut gio_sys::GIOStream,
+) -> *mut gio_sys::GInputStream
+where
+    T: IOStreamImpl,
+{
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: IOStream = from_glib_borrow(ptr);
+
+    let ret = imp.get_input_stream(&wrap);
+
+    // FIXME: We somehow need to ensure the stream stays alive
```

Comment:
This needs some thinking: we need to ensure that the returned stream stays alive as long as the instance. Could go via `GObject` qdata but that would easily cause circular references and thus memory leaks.

Needs some thinking.

Reply:
qdata would actually be fine for it in this case

Reply:
Good to hear that

Reply:
Unfortunately means that this specific one can't be autogenerated later :)